### ISSUE TITLE
Fetch experiments on startup first before applying

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/fenix#23758]
Duplicate of: https://github.com/mozilla-mobile/focus-android/issues/6443

When we want to run experiments on startup for new users. We need to make sure we apply the experiments only after fetching them.

We should land the same patch as Focus here or better, upstream these nuances into it's android component.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-19513)

**Comment by tarikeshaq** [comment]
Saw this passing by, some relevant notes from a previous discussion (and reasoning why it's currently the way it is): https://github.com/mozilla/application-services/blob/main/docs/adr/0004-early-startup-experiments.md
